### PR TITLE
🤖 Sync option descriptions <- website

### PR DESCRIPTION
**Pull request by jablko**
Get command-line option descriptions from the website Markdown and update their corresponding descriptions in the compiler.

/cc @orta

### CODE REVIEW

File: src/compiler/commandLineParser.ts [hunk 1]
```diff
@@ -728,7 +728,7 @@ namespace ts {
             type: "boolean",
             affectsSemanticDiagnostics: true,
             category: Diagnostics.Type_Checking,
-            description: Diagnostics.Interpret_optional_property_types_as_written_rather_than_adding_undefined,
```
Comment (jablko) [inline src/compiler/commandLineParser.ts:731]: Preferable? https://github.com/microsoft/TypeScript/pull/44626/files#diff-fdf3f532dc2d3188ad1c05c8c9e064d0aa4143b8ba09b8f3313a0243f0a43effR4856

File: src/compiler/commandLineParser.ts [hunk 2]
```diff
@@ -728,7 +728,7 @@ namespace ts {
             type: "boolean",
             affectsSemanticDiagnostics: true,
             category: Diagnostics.Type_Checking,
-            description: Diagnostics.Interpret_optional_property_types_as_written_rather_than_adding_undefined,
+            description: Diagnostics.Differentiate_between_undefined_and_not_present_when_type_checking,
```
Comment (jablko) [inline src/compiler/commandLineParser.ts:731]: https://github.com/microsoft/TypeScript-Website/pull/1906/commits/2689dd84bcfebf286099dd88e5cce68a836553ca#diff-00e96462280a2e50c9f8e9de40b441c06b040b6dd8ae51d284654cbefa3e5fb8R3

File: tests/baselines/reference/classStaticBlock22.errors.txt [hunk 3]
```diff
@@ -1,94 +0,0 @@
-tests/cases/conformance/classes/classStaticBlock/classStaticBlock22.ts(4,9): error TS1359: Identifier expected. 'await' is a reserved word that cannot be used here.
```
Comment (gabritto) [inline tests/baselines/reference/classStaticBlock22.errors.txt:1]: why have this and a bunch of other files been deleted?

Comment (jablko) [reply tests/baselines/reference/classStaticBlock22.errors.txt:1]: @gabritto Hi! They're baselines for tests that no longer exist --- they're removed by the Accept Baselines and Fix Lints workflow:
- https://github.com/microsoft/TypeScript/blob/6aaefe41fe03e4e67eda2f2c81df57adc08fd25b/.github/workflows/accept-baselines-fix-lints.yaml#L23
- https://github.com/microsoft/TypeScript/pull/45916

File: src/compiler/diagnosticMessages.json [hunk 4]
```diff
@@ -879,7 +879,6 @@
         "category": "Error",
         "code": 1271
     },
-
```
Comment (sandersn) [inline src/compiler/diagnosticMessages.json:882]: the whitespace here is a little useful for pointing out the gap between 1271 and 1300. I'd prefer to keep it and the others that are like it.

Comment (gabritto) [inline src/compiler/diagnosticMessages.json:882]: as Nathan commented, please leave the blank lines as-is

**Review by gabritto — CHANGES_REQUESTED** [review]
minor comment about empty lines, but other than that I think we're good to go

File: src/compiler/diagnosticMessages.json (hunk 4 above)
Comment (jablko) [reply src/compiler/diagnosticMessages.json]: - [x] @gabritto Thanks! I've restored the blank lines.

**Review by gabritto — APPROVED** [review]

**Pull request merged**